R/utilities-parameter.R

```diff
@@ -34,15 +42,58 @@ getAllParametersMatching <- function(path, container) {
 #' @export
 getParameter <- function(path, container) {
   parameters <- getAllParametersMatching(path, container)
-  stopifnot(length(parameters) <= 1)
+  if (length(parameters) > 1) {
+    stop(errorGetEntityMultipleOutputs(path, container))
+  }
+
   if (length(parameters) == 0) {
     return(NULL)
   }
 
-  parameters[[1]]
+  return(parameters[[1]])
 }
 
 toParameters <- function(netParams) {
   sapply(netParams, function(p)
     Parameter$new(p))
 }
+
+#' Set values of parameters
+#'
+#' @param parameters A single object of type 'Parameter' or a list of such objects
+#' @seealso \code{\link{getParameter}} and \code{\link{getAllParametersMatching}} to create objects of type Parameter
+#'
+#' @param values A numeric value that should be assigned to the parameter or a vector
+#' of numeric values, if the value of more than one parameter should be changed. Must have the same
+#' length as 'parameters'
+#'
+#' @examples
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' param <- getParameter(c("Organism", "Liver", "Volume"), sim)
+#' setParametersValues(param, 1)
+#' params <- getAllParametersMatching(c("Organism", "**", "Volume"), sim)
+#' setParametersValues(params, c(2, 3))
+#' @export
+setParametersValues <- function(parameters, values) {
+  # Must turn the input into a list so we can iterate through even when only
+  # one parameter is passed
+  parameters <- c(parameters)
+
+  # Test for correct inputs
+  if (!isOfType(parameters, "Parameter")) {

```

Comment:
What about we add some other validation methods 
such as validateIsType(), validateIsNumeric, validateIsSamelLength
That way you only call this methods 

validateIsNumeric would be 
```R
validateIsNumeric <-function(values) {
 if (!is.numeric(values)) {
    stop(errorWrongType(values))
  }
}
```

That way you don't repeat yourself. This something we will often probably
Thoughts?

Reply:
Yes this makes sense. However, this will lead to difficulties in generating error messages:
```

  # Name of the calling function
  callingFunctions <- sys.calls()
  callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
```

Because sometimes the error message function can be called from `validateIs`, and sometimes directly from the function we are interested in, so the the depth of the function we want to retrieve the name from (or the names of the arguments) will vary.

Reply:
oh i see. Ok let's forget about it for now if it's too complicated. 

Maybe there is something to be learned from https://adv-r.hadley.nz/conditions.html#call-stacks

Reply:
Yes, maybe I'll revisit it later.